parser/diff.go

```diff
@@ -104,6 +104,10 @@ func (p *DiffParser) Parse(r io.Reader) ([]*rdf.Diagnostic, error) {
 				}
 				prevState = diffLine.Type
 			}
+			if hunk.EOFNewline == diff.LineAdded {
+				state.newLines = append(state.newLines, "")
+				// TODO: also handle the much rarer case of a deleted eof newline
```

Comment:
[nit] I think we don't have a clear plan/TODO to support deleting EOF newline.
I prefer mark it as `NOTE:` or something because it's actually not in our TODO list.